apps/vor/pkg/school/school.go

```diff
@@ -83,7 +85,53 @@ func getClasses(server rest.Server, store postgres.SchoolStore) http.Handler {
 		return nil
 	})
 }
+func getClassAttendance(server rest.Server, store postgres.SchoolStore) http.Handler {
+	type attendanceData struct {
+		StudentId string `json:"studentId"`
+		Name      string `json:"name"`
+		Attend    bool   `json:"attend"`
+	}
 
+	return server.NewHandler(func(w http.ResponseWriter, r *http.Request) *rest.Error {
+		session := chi.URLParam(r, "session")
+		classId := chi.URLParam(r, "classId")
+		attendance, err := store.GetClassAttendance(classId, session)
+		if err != nil {
+			return &rest.Error{
+				Code:    http.StatusInternalServerError,
+				Message: "failed querying attendance",
+				Error:   err,
+			}
+		}
+		var response []attendanceData
+		attendMap := make(map[string]int)
+		if len(attendance) > 0 {
+			students := attendance[0].Class.Students
```

Comment:
Ouw cool, that's pretty efficient, getting the class's student data by joining the attendance table with the class table. I was expecting two separate queries lol

Reply:
Thanks chief @chrsep